--- a/src/nautilus-files-view.c
+++ b/src/nautilus-files-view.c
@@ -5855,18 +5855,24 @@
                               gpointer       user_data)
 {
     NautilusFilesView *self = user_data;
+    GFile *location;
+    g_autofree char *text = NULL;
     GdkClipboard *clipboard;
-    GList *files;
 
-    if (self->directory_as_file != NULL)
+    location = nautilus_files_view_get_location (self);
+    if (location == NULL)
     {
-        files = g_list_append (NULL, nautilus_file_ref (self->directory_as_file));
-
-        clipboard = gtk_widget_get_clipboard (GTK_WIDGET (self));
-        nautilus_clipboard_prepare_for_files (clipboard, files, FALSE);
+        return;
+    }
 
-        nautilus_file_list_free (files);
+    text = g_file_get_path (location);
+    if (text == NULL)
+    {
+        text = g_file_get_uri (location);
     }
+
+    clipboard = gtk_widget_get_clipboard (GTK_WIDGET (self));
+    gdk_clipboard_set_text (clipboard, text);
 }
 
 static void
